### PR TITLE
Continuous Integration Support for MATLAB using Travis CI

### PR DESCRIPTION
**Pull request by jwpereira**
Using an ephemeral instance of MATLAB®, projects using a .travis.yml configuration of

```yaml
language: matlab
```

can easily have their MATLAB and Simulink® tests run on Travis CI.

### CODE REVIEW

File: lib/travis/build/script/matlab.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+module Travis
+    module Build
+      class Script
+        class Matlab < Script
+          MATLAB_INSTALLER_LOCATION = 'https://ssd.mathworks.com/supportfiles/ci/ephemeral-matlab/v0/install.sh'.freeze
+          MATLAB_EXECUTABLE = 'matlab -batch'.freeze
+          MATLAB_COMMAND = "results = runtests('IncludeSubfolders',true," \
+                              "'IncludeReferencedProjects',true), " \
+                              'assert(all(~[results.Failed]))'.freeze
```
Comment (acampbel) [inline lib/travis/build/script/matlab.rb]: DO we think that the installer will install 20a by the time this gets merged? If so, we should change this line to:

`"assertSuccess(runtests('IncludeSubfolders',true,'IncludeReferencedProjects',true)"`

File: lib/travis/build/script/matlab.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+module Travis
+    module Build
+      class Script
+        class Matlab < Script
+          MATLAB_INSTALLER_LOCATION = 'https://ssd.mathworks.com/supportfiles/ci/ephemeral-matlab/v0/install.sh'.freeze
+          MATLAB_EXECUTABLE = 'matlab -batch'.freeze
+          MATLAB_COMMAND = "results = runtests('IncludeSubfolders',true," \
+                              "'IncludeReferencedProjects',true), " \
```
Comment (mcafaro) [inline lib/travis/build/script/matlab.rb]: Think we decided against using `'IncludeReferencedProjects'` by default.

**Review by acampbel — APPROVED** [review]

**Review by mcafaro — APPROVED** [review]
Looking good! Just need to get the default command settled.

File: lib/travis/build/script/matlab.rb (hunk 2 above)
Comment (acampbel) [reply lib/travis/build/script/matlab.rb]: even easier then:
`assertSuccess(runtests('IncludeSubfolders',true))`

Comment (jwpereira) [reply lib/travis/build/script/matlab.rb]: Sounds good! I will go ahead and use that command then :)

File: lib/travis/build/script/matlab.rb (hunk 1 above)
Comment (jwpereira) [reply lib/travis/build/script/matlab.rb]: Going with your suggestion under @mcafaro's comment :)

**Comment by jwpereira** [comment]
Okay! Along with some minor touchups to the code itself (no idea how I missed the indentation being completely off 😮), I went ahead and updated the code to use `assertSuccess` from R2020a 😄

File: lib/travis/build/script/matlab.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+module Travis
+  module Build
+    class Script
+      class Matlab < Script
+        MATLAB_INSTALLER_LOCATION = 'https://ssd.mathworks.com/supportfiles/ci/ephemeral-matlab/v0/install.sh'.freeze
+        MATLAB_START = 'matlab -batch'.freeze
+        MATLAB_COMMAND = "results = assertSuccess(runtests('IncludeSubfolders',true))".freeze
```
Comment (mcafaro) [inline lib/travis/build/script/matlab.rb]: I think we can do without assigning to `results` and just have `MATLAB_COMMAND = "assertSuccess(runtests('IncludeSubfolders',true));".freeze` (with a semicolon at the end). If `assertSuccess` fails it throws an exception, in which case results won't be returned anyway. If it succeeds, results will be returned and displayed but that will be inconsistent with the failure case.

Comment (jwpereira) [reply lib/travis/build/script/matlab.rb]: Ohhh I see. That makes a lot of sense. I went ahead and updated the command to reflect that 😄 
(If you're wondering about the force-push.. I almost forgot the semicolon) 🤣

**Comment by mcafaro** [comment]
Ship it!

**Comment by jwpereira** [comment]
Closing this one out for housekeeping. Now the ball is over at https://github.com/travis-ci/travis-build/pull/1880

**Pull request closed without merging**